### PR TITLE
some landscape fixes

### PR DESCRIPTION
**Pull request by scopatz**
Not ready yet.  Just using the PR to trigger landscape.

### CODE REVIEW

**Comment by scopatz** [comment]
OK, I think this is ready to merge.  Landscape it being slow, but this definitely improves our score.

File: xonsh/aliases.py [hunk 1]
```diff
@@ -54,7 +53,7 @@ def source_bash(args, stdin=None):
     denv = env.detype()
     with tempfile.NamedTemporaryFile(mode='w+t') as f:
         args = ' '.join(args)
-        input = 'source {0}\nenv >> {1}\n'.format(args, f.name)
+        inp = 'source {0}\nenv >> {1}\n'.format(args, f.name)
         try:
             subprocess.check_output(['bash'],
                                     input=input,
```
Comment (wrywerytwreywery) [inline xonsh/aliases.py]: Should now be `input=inp,`

File: xonsh/completer.py [hunk 2]
```diff
@@ -133,7 +131,7 @@ def bash_complete(self, prefix, line, begidx, endidx):
         fnme = self.bash_complete_files.get(cmd, None)
         if func is None or fnme is None:
             return set()
-        idx = 0
+        idx = n = 0
```
Comment (wrywerytwreywery) [inline xonsh/completer.py]: What is the rationale behind this change?  The loop that follows will assign `n` anyway, no?

File: xonsh/parser.py [hunk 3]
```diff
@@ -634,10 +633,10 @@ def p_varargslist(self, p):
         p1, p2 = p[1], p[2]
         p3 = p[3] if lenp > 3 else None
         p4 = p[4] if lenp > 4 else None
-        p5 = p[5] if lenp > 5 else None
+        #p5 = p[5] if lenp > 5 else None
         p6 = p[6] if lenp > 6 else None
         p7 = p[7] if lenp > 7 else None
-        p8 = p[8] if lenp > 8 else None
+        #p8 = p[8] if lenp > 8 else None
```
Comment (wrywerytwreywery) [inline xonsh/parser.py]: If we want to delete these, should we just delete them (rather than commenting out)?

**Comment by wrywerytwreywery** [comment]
I just took a quick look; a few questions/comments in-line.

File: xonsh/completer.py (hunk 2 above)
Comment (scopatz) [inline xonsh/completer.py]: `n` is used outside of the for-loop below, and so if by some chance `splt` is length-0, then this prevents a name lookup error.  I agree that this is pretty obtuse, but it was one of the things that landscape was yelling about.  It is better and easier to 'fix' it than to suppress the error.

File: xonsh/parser.py (hunk 3 above)
Comment (scopatz) [inline xonsh/parser.py]: likely :) On it now.

**Comment by scopatz** [comment]
Thanks for the review @wrywerytwreywery. I think this is ready for another look.

File: xonsh/completer.py (hunk 2 above)
Comment (wrywerytwreywery) [inline xonsh/completer.py]: Ah, alright; I didn't catch that.

**Comment by wrywerytwreywery** [comment]
Looks good to me!  Merging in.
